max-package/jsui/fluid.waveform~.js

```diff
@@ -378,55 +383,68 @@ function addlayer(type, source, r, g, b, a)
     refresh();
 }
 
+// Different methods for adding slices
+// They all call addmarkers internally
+function slices(source, reference)
+{
+    addmarkers(source, reference);
+}
+
+function markers(source, reference)
+{
+    addmarkers(source, reference);
+}
+
 function indicesbuffer(source, reference)
 {
-    if (typeof source === 'string')
-    {
-        addmarkers(source, reference);
-    }
+    addmarkers(source, reference);
 }
 
 function addmarkers(source, reference)
 {
-    if (!source)
+    // Check that the correct arguments are supplied
+    if (!source && typeof source !== "string")
     {
         err('marker layer must have a source (buffer)');
         return
     }
 
-    if (!bufexists(source))
-    {
-        err('buffer' + ' "' + source + '" ' + 'does not exist');
-        return
-    }
-
-    if (bufempty(source))
+    if (!reference)
     {
-        err('buffer' + ' "' + source + '" ' + 'is empty');
+        err('no reference buffer or sample rate provided');
         return
     }
 
-    if (badslicebuf(source))
-    {
-        err('slices buffer has no valid slice points');
-        return
+    if (typeof reference === "string") {
+        post('doing string')
```

Comment:
yikes

Reply:
hang on, where is the numeric case handled?

Reply:
You only need to handle the case that your reference is a string (buffer name) as the using of that reference happens further down in the MarkerSpec.